### ISSUE TITLE
Feature request: Support fetching the transaction ID for not-yet-mined transactions

### ISSUE THREAD

**Issue by Arachnid** [trufflesuite/truffle#652]
At present, the promise returned from contract functions only resolves when the transaction is mined. To provide users with more insight into transaction processing, it'd be ideal if it were possible to fetch the transaction ID as soon as it's submitted, instead of only when it's mined.

It's possible to explicitly call `sendTransaction` on a contract method, which resolves as soon as the transaction is submitted, but no functionality is exposed to then provide a promise that resolves when the transaction is later mined.

**Comment by cgewecke** [comment]
This feature is available as part of an upgrade to `web3 1.0` which is now available in an experimental nightly build here: [darq-truffle@next](https://www.npmjs.com/package/darq-truffle). 
```shell
$ npm install -g darq-truffle@next
$ darq-truffle test # Example command
```

Usage docs [here](https://github.com/trufflesuite/truffle-contract/blob/web3-one-readme/README.md)

**Comment by gnidan** [comment]
This should be fixed in v5 via the `contract.myFunction().on('transactionHash', () => { ... })` mechanism. Thanks for raising the issue!
